addons/stock/models/stock_picking.py

```diff
@@ -806,7 +805,7 @@ def create(self, vals_list):
         for vals in vals_list:
             defaults = self.default_get(['name', 'picking_type_id'])
             picking_type = self.env['stock.picking.type'].browse(vals.get('picking_type_id', defaults.get('picking_type_id')))
-            if vals.get('name', '/') == '/' and defaults.get('name', '/') == '/' and vals.get('picking_type_id', defaults.get('picking_type_id')):
+            if vals.get('name', '/') == '/' and defaults.get('name', 'New') == _('New') and vals.get('picking_type_id', defaults.get('picking_type_id')):
```

Comment:
Do not compare translatable strings with hardcoded ones. You can rather do like this : 

https://github.com/odoo/odoo/blob/efe6dac4a1757da935584536f6acb1f6d14eb915/addons/mrp/models/mrp_production.py#L908